cmd/juju/commands/bootstrap.go

```diff
@@ -598,7 +607,7 @@ func (c *bootstrapCommand) handleCommandLineErrorsAndInfoRequests(ctx *cmd.Conte
 	return false, nil
 }
 
-func (c *bootstrapCommand) cloud(ctx *cmd.Context) (*jujucloud.Cloud, error) {
+func (c *bootstrapCommand) cloud(ctx *cmd.Context) (string, *jujucloud.Cloud, error) {
```

Comment:
It seems a little odd that "cloudName" takes precedence over the Cloud object.

Also, I probably would have thought you would do
 cloud.Name()
at some later time, rather than having this 'cloud' object do the indirection to find the preferred default name.
 (or possibly DefaultCloudName(cloudObject))
Then you could even just do:
 c.Cloud = DefaultCloudName(cloudObject)
and the other code wouldn't have to change?